### PR TITLE
Log how much time it takes e2e tests to clean up the namespace

### PR DESCRIPTION
**Pull request by smarterclayton**
Will get a better handle on deletion test wasted time

### CODE REVIEW

**Comment by smarterclayton** [comment]
Wow... namespace deletion on a lot of tests is taking ~30s.  That's a crazy amount of wasted time.  I think most of that is explicit deletion time on all the pods in the namespace, will experiment.

**Comment by smarterclayton** [comment]
[times.txt](https://github.com/kubernetes/kubernetes/files/839913/times.txt)

was graphed into 

https://docs.google.com/spreadsheets/d/1C_NwOUKUYwxwKqXe-Zz4iCq17ITGDwOQlGnvtTcyBHk/edit?usp=sharing

**Comment by smarterclayton** [comment]
Strong central trend line looks like rate limiting on the namespace controller - we run out of throttled time.

**Comment by rmmh** [comment]
Did you want to remove the debug termination timelimits and get the logging in?

**Comment by smarterclayton** [comment]
Probably although no rush right now.  I was hoping this was something
obvious (rate limit, long termination periods, or something), and it's not
(it's a combination of things, or I'm not finding the right rate limit).

**Comment by smarterclayton** [comment]
For historical context, taking the rate limiter off the namespace controller did not improve this.

File: cmd/kube-controller-manager/app/core.go [hunk 1]
```diff
@@ -104,7 +104,10 @@ func startNamespaceController(ctx ControllerContext) (bool, error) {
 
 	// Find the list of namespaced resources via discovery that the namespace controller must manage
 	namespaceKubeClient := ctx.ClientBuilder.ClientOrDie("namespace-controller")
-	namespaceClientPool := dynamic.NewClientPool(ctx.ClientBuilder.ConfigOrDie("namespace-controller"), restMapper, dynamic.LegacyAPIPathResolverFunc)
+	pool := ctx.ClientBuilder.ConfigOrDie("namespace-controller")
```
Comment (deads2k) [inline cmd/kube-controller-manager/app/core.go]: missed the other client

**Comment by deads2k** [comment]
@smarterclayton can you wire https://github.com/kubernetes/kubernetes/pull/44455 into your test?

**Comment by smarterclayton** [comment]
Yeah.

**Comment by smarterclayton** [comment]
Testing now

**Comment by smarterclayton** [comment]
Going to push this to help debug  #45705

**Comment by smarterclayton** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by smarterclayton** [comment]
@ncdc included your logging error you saw

**Comment by smarterclayton** [comment]
Self applying to get some debug info

/lgtm

**Comment by rmmh** [comment]
You want to merge this with the new termination timelimits?

**Comment by smarterclayton** [comment]
Yes, those are teardowns and safe.

**Pull request merged**